File: backend/app/Models/ProductionLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ProductionLog extends Model
{
    protected $fillable = [
        'production_order_id', 'user_id', 'date', 'shift',
        'produced_quantity', 'notes',
    ];

    protected $casts = [
        'date'              => 'date',
        'produced_quantity' => 'integer',
    ];

    public function productionOrder()
    {
        return $this->belongsTo(ProductionOrder::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: backend-dotnet/Models/ProductionLog.cs
namespace TexTrack.Api.Models;

public class ProductionLog
{
    public long Id { get; set; }
    public long ProductionOrderId { get; set; }
    public long? UserId { get; set; }
    public DateOnly Date { get; set; }
    public string Shift { get; set; } = string.Empty; // sabah | öğle | gece
    public int ProducedQuantity { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ProductionOrder? ProductionOrder { get; set; }
    public User? User { get; set; }
}
